pkg/maintenance/postgresql.go

```diff
@@ -102,15 +119,29 @@ func (p *PostgreSQL) DoMaintenance(ctx context.Context) error {
 	}
 
 	p.log.Info("Repacking databases...")
-	err = p.createRepack(clusterName)
+	err = p.createRepack(sgCluster.GetName())
 	if err != nil {
 		return fmt.Errorf("cannot create repack: %v", err)
 	}
 
 	return nil
 }
 
-func (p *PostgreSQL) upgradeVersion(currentVersion string, clusterName string) (OpName, error) {
+func (p *PostgreSQL) checkRestart(sgCluster stackgresv1.SGCluster) bool {
```

Comment:
`checkRestart` seems like a misnomer. We don't check for the restart condition (which is actually a seperate condition that exists), but we check the `ClusterRequiresUpgrade` condition.

So maybe `checkRequiredUpgrade`?

Reply:
Good point. Renamed the function and the related Log output